### DISCUSSION TITLE
Update model relation

### DISCUSSION THREAD

**Discussion by Vishwas-verma** [adonisjs/core#2832] [General]
Hye,
```
    if (data.identity) {
      const identity = (await driver.related("identity")) as unknown as DriverIdentity;
      await identity.merge({...data.identity}).save();
    }
```
Driver and Identity has one to one relation .
Presently I am forcefully converting relationship Instance to LucidRow and then applying merge but it seems its not converting to LucidRow,how to do it so that I can use merge ?

**Error**
```
identity.merge is not a function
```

Is there any other way to update relationship model in adonisJS ?

**Comment by bitkidd** [comment]
When you want to query related model, you should do something like this:
```ts
const identity = await driver.related('identity').query()
```
This will return you an array of related models, the first one is yours, you can call something like this:
```ts
const identity = await driver.related('identity').query().first()
```
This will return just one instance.
Then you can do `merge` and `save`.

More docs info [here](https://docs.adonisjs.com/guides/models/relationships#relationship-query-builder).

**Reply by Vishwas-verma** [reply]
Hye @bitkidd ,
I did that after doing this the value gets updated in database but in response I am getting previous value of identity.

Do I have to set value of idenity to driver or is there any better way.

For Example I am currently doing this to update relation-
```
  async update(driver: Driver, data: DriverUpdateDto): Promise<Driver> {
    let updatedIdentity;
    if (data.identity) {
      const identity = await driver.related("identity").query().first();
      if (identity) updatedIdentity = await identity.merge({ ...data.identity }).save();
    }
    let updatedCab;
    if (data.cab) {
      const cab = await driver.related("cab").query().first();
      if (cab) updatedCab = await cab.merge({ ...data.cab, driverId: driver.id }).save();
    }
    const updatedDriver = await driver.merge({ ...data }).save();
    await updatedDriver.$setRelated("identity", updatedIdentity);
    await updatedDriver.$setRelated("cab", updatedCab);
    return updatedDriver;
  }
```

Is there any better way than this in time space tradeOff ??
